pkgs/by-name/ks/ksoloti/package.nix

```
@@ -0,0 +1,181 @@
+{ ant
+, copyDesktopItems
+, dfu-util
+, fetchpatch
+, fetchurl
+, fetchFromGitHub
+, gcc-arm-embedded-6
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenv
+, pkg-config
+, systemd
+, unzip
+, zulu
+,
+}:
+let
+  ksolotiVersion = "1.0.12-8";
+  libusbVersion = "1.0.19";
+  chibiOSVersion = "2.6.9";
+
+  src = fetchFromGitHub {
+    owner = "ksoloti";
+    repo = "ksoloti";
+    rev = ksolotiVersion;
+    sha256 = "sha256-UYxCnEZORpfhaHSl6YohuFiQLPoQ1oyJUL9W6Maf8AE=";
+  };
+
+  libusb = stdenv.mkDerivation rec {
+    name = "libusb";
+    version = libusbVersion;
+    src =
+      let
+        minorVersion = builtins.concatStringsSep "." (
+          lib.take 2 (builtins.splitVersion version)
+        );
+      in
+      fetchurl {
+        url = "https://sourceforge.net/projects/libusb/files/libusb-${minorVersion}/libusb-${version}/libusb-${version}.tar.bz2";
+        sha256 = "sha256-bFAsgWAC+Q1PdgUKZCnDp+DYQgQiLL/y3Old13O6aEA=";
+      };
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ systemd ];
+
+    patches = [
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/ksoloti/ksoloti/${ksolotiVersion}/platform_linux/src/libusb.stdfu.patch";
+        sha256 = "sha256-bYzcE20fnwBTZUZmYYHgkW7SekO6pgoj6AaTGIw8kqQ=";
+      })
+    ];
+  };
+
+  patched-dfu-util = dfu-util.overrideAttrs (oldAttrs: {
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ libusb ];
+  });
+
+  chibios = stdenv.mkDerivation {
+    name = "chibios";
+    version = chibiOSVersion;
+    src = fetchFromGitHub {
+      owner = "ChibiOS";
+      repo = "ChibiOS";
+      rev = "ver${chibiOSVersion}";
+      hash = "sha256-ILb6OluhriztSRbVIqb4NGzmbSlH9cb+NOw3b58Hz0I=";
+    };
+
+    buildInputs = [ unzip ];
+
+    patchPhase = ''
+      cd ext
+      unzip -o fatfs-0.9-patched.zip
+      cd ..
+    '';
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir -p $out
+      cp -r * $out
+    '';
```

phases should (almost always) run their pre- and post- hooks

```suggestion
    patchPhase = ''
      runHook prePatch
    
      cd ext
      unzip -o fatfs-0.9-patched.zip
      cd ..
      
      runHook postPatch
    '';

    dontBuild = true;

    installPhase = ''
      runHook preInstall
    
      mkdir -p $out
      cp -r * $out
      
      runHook postInstall
    '';
```

---

ah great! i was wondering why the postInstall hook wasn't running. this explains it. :+1: